tests/lean/run/invalidProjection.lean

```diff
@@ -0,0 +1,104 @@
+import Lean.Parser.Term
+
+/-!
+# Invalid projection error messages
+
+This file assesses error messages produced for invalid projections.
+-/
+
+inductive P : Nat → n > 0 → Prop
+  | mk (n) (q : n > 0) : P n q
+
+/--
+error: Invalid projection: Index `2` is invalid for this structure; the only valid index is 1
+
+Note: The expression
+  h2
+has type
+  P m h'
+which has only 1 field
+-/
+#guard_msgs in
+example (h1 : P n h) (h2 : P m h') := h1.1 = h2.2
+
+/--
+error: Invalid projection: Index `3` is invalid for this structure; it must be between 1 and 2
+
+Note: The expression
+  x
+has type
+  Nat × Nat × Nat
+which has only 2 fields
+-/
+#guard_msgs in
+example (x : Nat × Nat × Nat) := x.3
```

Comment:
This would be a cherry on top of the cake, but probably help the haskellers: Detect when the user is facing nested pairs, which look like n-ary tuples during construction, and add a Note that this is only superficial syntax, but cannot be used with projections.
(no need to do here, but maybe create an issue for a rainy day.)